direct/data/transforms.py

```diff
@@ -401,10 +401,66 @@ def complex_multiplication(input_tensor: torch.Tensor, other_tensor: torch.Tenso
         ],
         dim=complex_index,
     )
-
     return multiplication
 
 
+def complex_dot_product(a: torch.Tensor, b: torch.Tensor, dim: List[int]) -> torch.Tensor:
+    r"""Computes the dot product of the complex tensors :math:`a` and :math:`b`: :math:`a^{*}b = <a, b>`.
+
+    Parameters
+    ----------
+    a : torch.Tensor
+        Input :math:`a`.
+    b : torch.Tensor
+        Input :math:`b`.
+    dim : List[int]
+        Dimensions which will be suppressed. Useful when inputs are batched.
+
+    Returns
+    -------
+    complex_dot_product : torch.Tensor
+        Dot product of :math:`a` and :math:`b`.
+    """
+    return complex_multiplication(conjugate(a), b).sum(dim)
```

Comment:
Can you not do `view_as_complex` and then `view_as_real`? Do a check `is_complex`, and if so, you just multiply. Otherwise, you `view_as_complex` and multiply. Need to check if that works, because perhaps some pytorch operations would not like that (it not being contiguous or so)

Reply:
PLease create an issue and could do it in future commits.